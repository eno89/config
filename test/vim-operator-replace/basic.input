" Tests for operator-replace
function s:test_named_key_mappings()  "{{{1
  Title "Named key mappings"

  let _ = '<Plug>(operator-replace)'
  Assert maparg(_, 'c') != '' ==> 0
  Assert maparg(_, 'i') != '' ==> 0
  Assert maparg(_, 'n') != '' ==> 1
  Assert maparg(_, 'o') != '' ==> 1
  Assert maparg(_, 'v') != '' ==> 1

  Assert maparg(_, 'n') =~# 'operator#replace#' ==> 1
  Assert maparg(_, 'o') ==> 'g@'
  Assert maparg(_, 'v') =~# 'operator#replace#' ==> 1
endfunction




function s:test_usage_block_vs_block()  "{{{1
  Title "Usage: block {motion} vs block {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_block_vs_char()  "{{{1
  Title "Usage: block {motion} vs char {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_block_vs_line()  "{{{1
  Title "Usage: block {motion} vs line {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_char_vs_block()  "{{{1
  Title "Usage: char {motion} vs block {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_char_vs_char()  "{{{1
  Title "Usage: char {motion} vs char {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY
endfunction




function s:test_usage_char_vs_line()  "{{{1
  Title "Usage: char {motion} vs line {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_line_vs_block()  "{{{1
  Title "Usage: line {motion} vs block {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_line_vs_char()  "{{{1
  Title "Usage: line {motion} vs char {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY - but what result I expect for this case?
endfunction




function s:test_usage_line_vs_line()  "{{{1
  Title "Usage: line {motion} vs line {register} content"

  tabnew
  tabonly!

  let text = [
  \     '(herb oil pepper) salt water',
  \     'Super Super Mega Kawaii',
  \     'Sasara moe!  It''s burning!'
  \   ]
  silent put =text
  1 delete _
  Assert getline(1, line('$')) ==> text

  " FIXME: NIY
endfunction




Test  "{{{1

" __END__
" vim: filetype=vim foldmethod=marker
