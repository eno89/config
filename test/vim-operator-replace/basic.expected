===== "Named key mappings"
TEST: maparg(_, 'c') != ''  ==>  0 ... ok
TEST: maparg(_, 'i') != ''  ==>  0 ... ok
TEST: maparg(_, 'n') != ''  ==>  1 ... ok
TEST: maparg(_, 'o') != ''  ==>  1 ... ok
TEST: maparg(_, 'v') != ''  ==>  1 ... ok
TEST: maparg(_, 'n') =~# 'operator#replace#'  ==>  1 ... ok
TEST: maparg(_, 'o')  ==>  'g@' ... ok
TEST: maparg(_, 'v') =~# 'operator#replace#'  ==>  1 ... ok
Result: 8 / 8

===== "Usage: block {motion} vs block {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: block {motion} vs char {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: block {motion} vs line {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: char {motion} vs block {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: char {motion} vs char {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
TEST: [line('.'), col('.')]  ==>  [1, 7] ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed', 'v'] ... ok
TEST: getline(1)  ==>  '(herb unnamed pepper) salt water' ... ok
TEST: getline(2)  ==>  'Super Super Mega Kawaii' ... ok
TEST: getline(3)  ==>  'Sasara moe!  It''s burning!' ... ok
TEST: getline(4)  ==>  'Alter, you are also SSMK.' ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed', 'v'] ... ok
TEST: [line('.'), col('.')]  ==>  [2, 8] ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed2', 'v'] ... ok
TEST: getline(1)  ==>  '(herb unnamed pepper) salt water' ... ok
TEST: getline(2)  ==>  'Super Sunnamed2ga Kawaii' ... ok
TEST: getline(3)  ==>  'Sasara moe!  It''s burning!' ... ok
TEST: getline(4)  ==>  'Alter, you are also SSMK.' ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed2', 'v'] ... ok
TEST: [line('.'), col('.')]  ==>  [3, 8] ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed3', 'v'] ... ok
TEST: [getreg('s'), getregtype('s')]  ==>  ['!@#$%^&*()', 'v'] ... ok
TEST: getline(1)  ==>  '(herb unnamed pepper) salt water' ... ok
TEST: getline(2)  ==>  'Super Sunnamed2ga Kawaii' ... ok
TEST: getline(3)  ==>  'Sasara !@#$%^&*()!  It''s burning!' ... ok
TEST: getline(4)  ==>  'Alter, you are also SSMK.' ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed3', 'v'] ... ok
TEST: [getreg('s'), getregtype('s')]  ==>  ['!@#$%^&*()', 'v'] ... ok
TEST: [line('.'), col('.')]  ==>  [4, 21] ... ok
TEST: [getreg('s'), getregtype('s')]  ==>  [')(*&^%$#@!', 'v'] ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed4', 'v'] ... ok
TEST: getline(1)  ==>  '(herb unnamed pepper) salt water' ... ok
TEST: getline(2)  ==>  'Super Sunnamed2ga Kawaii' ... ok
TEST: getline(3)  ==>  'Sasara !@#$%^&*()!  It''s burning!' ... ok
TEST: getline(4)  ==>  'Alter, you are also )(*&^%$#@!.' ... ok
TEST: [getreg('"'), getregtype('"')]  ==>  ['unnamed4', 'v'] ... ok
TEST: [getreg('s'), getregtype('s')]  ==>  [')(*&^%$#@!', 'v'] ... ok
Result: 33 / 33

===== "Usage: char {motion} vs line {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: line {motion} vs block {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: line {motion} vs char {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== "Usage: line {motion} vs line {register} content"
TEST: getline(1, line('$'))  ==>  text ... ok
Result: 1 / 1

===== Total
Result: 49 / 49
